src/agentscope/rag/llama_index_knowledge.py

```diff
@@ -0,0 +1,579 @@
+        # let the doc_id be the filename for each document
+        loader.filename_as_id = True
+        if query is None:
+            documents = loader.load_data()
+        else:
+            # this is for querying a database,
+            # does not work for loading a document directory
+            documents = loader.load_data(query)
+        logger.info(f"loaded {len(documents)} documents")
+        return documents
+
+    def _docs_to_nodes(
+        self,
+        documents: List[Document],
+        transformations: Optional[list[TransformComponent]] = None,
+    ) -> Any:
+        """
+        Convert the loaded documents to nodes using transformations.
+
+        Args:
+            documents (List[Document]):
+                documents to be processed, usually expected to be in
+                 llama index Documents.
+            transformations (Optional[list[TransformComponent]]):
+                optional, specifies the transformations (operators) to
+                process documents (e.g., split the documents into smaller
+                chunks)
+        Return:
+            Any: return the index of the processed document
+        """
+        # nodes, or called chunks, is a presentation of the documents
+        # we build nodes by using the IngestionPipeline
+        # for each document with corresponding transformations
+        pipeline = IngestionPipeline(
+            transformations=transformations,
+        )
+        # stack up the nodes from the pipline
+        nodes = pipeline.run(
+            documents=documents,
+            show_progress=self.showprogress,
+        )
+        logger.info("nodes generated.")
+        return nodes
+
+    def _set_loader(self, config: dict) -> Any:
+        """
+        Set the loader as needed, or just use the default setting.
+
+        Args:
+            config (dict): a dictionary containing configurations
+        """
+        if "load_data" in config:
+            # we prepare the loader from the configs
+            loader = self._prepare_args_from_config(
+                config=config.get("load_data", {}),
+            )
+        else:
+            # we prepare the loader by default
+            try:
+                from llama_index.core import SimpleDirectoryReader
+            except ImportError as exc_inner:
+                raise ImportError(
+                    " LlamaIndexAgent requires llama-index to be install."
+                    "Please run `pip install llama-index`",
+                ) from exc_inner
+            loader = {
+                "loader": SimpleDirectoryReader(
+                    input_dir="set_default_data_path",
+                ),
+            }
+        logger.info("loaders are ready.")
+        return loader
+
+    def _set_transformations(self, config: dict) -> Any:
+        """
+        Set the transformations as needed, or just use the default setting.
+
+        Args:
+            config (dict): a dictionary containing configurations
```

Comment:
Not an issue, since the function won't expose to users, but it would be better if the `config` arg is more specified( e.g. the store_and_index field required?)